### PR TITLE
Added archive core files for all platforms except zos

### PR DESCRIPTION
**Pull request by LongyuZhang**
- Added archive core files for all non-zos platforms
Issue: AdoptOpenJDK/openjdk-tests/issues/1564

Signed-off-by: Longyu Zhang <longyu.zhang@ibm.com>

### CODE REVIEW

**Comment by LongyuZhang** [comment]
Example for test failure is Grinder 304;
Example for build failure is tested locally with the output as follows:
```
     [exec] JVMDUMP039I Processing dump event "vmstop", detail "#0000000000000000" at 2020/04/05 22:35:20 - please wait.
     [exec] JVMDUMP032I JVM requested System dump using '/cores/core.20200405.223520.27476.0001.dmp' in response to an event
     [exec] JVMDUMP010I System dump written to /cores/core.20200405.223520.27476.0001.dmp
     [exec] JVMDUMP013I Processed dump event "vmstop", detail "#0000000000000000".

BUILD FAILED
/Users/lon/Documents/repos/openj9/test/TKG/scripts/build_test.xml:58: The following error occurred while executing this line:
/Users/lon/Documents/repos/openj9/test/functional/TestExample/build.xml:75: The following error occurred while executing this line:
/Users/lon/Documents/repos/openj9/test/functional/TestExample/build.xml:52: exec returned: 1

Total time: 22 seconds
1
20200405-22:35:42 - Attempting to move the TDUMPS to '/Users/lon/Documents/repos/openj9/test/TKG/../TKG/test_output_15861405191368/compile'.
original cores folder: total 14706280
 drwxr-xr-x   3 root  wheel          96  5 Apr 22:35 .
 drwxr-xr-x  22 root  admin         704  7 Mar 22:37 ..
 -r--------   1 root  wheel  7519457280  5 Apr 22:35 core.20200405.223520.27476.0001.dmp
 
20200405-22:35:42 - Successfully moved core file core.20200405.223520.27476.0001.dmp.
dest files: total 14706296
 drwxr-xr-x  4 root  staff         128  5 Apr 22:35 .
 drwxr-xr-x  3 root  staff          96  5 Apr 22:35 ..
 -rw-r--r--  1 root  staff        7608  5 Apr 22:35 compilation.log
 -r--------  1 root  wheel  7519457280  5 Apr 22:35 core.20200405.223520.27476.0001.dmp
```

**Comment by llxia** [comment]
Please fix the conflicts. Thanks.

File: scripts/moveDmp.pl [hunk 1]
```diff
@@ -126,98 +126,122 @@ sub moveTDUMPS {
...
-		my $deleteLog = "delete.log";
-		if ($dump !~ /X&DS/) {
-			my $coreDump = "core."."$dump".".dmp";
-			my $cmd = "mv \"//'${dump}'\" ". catfile($moveLocation, $coreDump);
-			qx($cmd 2>$moveLog);
-			($cmdSucceed, $dumpFound, $cleanFile) = checkLog($dump, $moveLog);
-			if ($cmdSucceed) {
-				logMsg("Found TDUMP named $dump, and renamed it $coreDump located at $moveLocation");
-				$movedDumps{$dump} = 1;
+	} elsif ($spec =~ /zos/) {
```
Comment (llxia) [inline scripts/moveDmp.pl:152]: This is PR for mac, we should not change the code for other platforms.

**Comment by LongyuZhang** [comment]
Example of build failure with default core dump location (moved): Grinder_TKG 334;
Example of build failure with core dump to /cores (moved): Grinder_TKG 338;
Example of test failure with default core dump location (no need move): Grinder_TKG 337.

File: scripts/moveDmp.pl [hunk 2]
```diff
@@ -127,8 +128,33 @@ sub moveTDUMPS {
 	# Use a hash to ensure that each dump is only dealt with once
 	my %parsedNames = ();
 	if ($spec !~ /zos/) {
-		my $moveCMD = "find ".${testRoot}." -name 'core.*.dmp' -exec mv -t ".${moveLocation}." '{}' +";
-		qx($moveCMD);
+		logMsg("Attempting to move core dumps to '$moveLocation'.");
```
Comment (llxia) [inline scripts/moveDmp.pl:131]: I don't think we need msg here.

File: scripts/moveDmp.pl [hunk 3]
```diff
@@ -127,8 +128,33 @@ sub moveTDUMPS {
...
-		qx($moveCMD);
+		logMsg("Attempting to move core dumps to '$moveLocation'.");		
+		if($file) {
+			while ($file =~ /System dump written to (.*)/g) {
+				my $curCorePath = $1;
+				my $curCoreName = basename($curCorePath);
+				if(!exists $parsedNames{$curCoreName}) {
+					# handle each core only once
+					$parsedNames{$curCoreName} = 1;
+					# If core not archived in preferred location, move it
```
Comment (llxia) [inline scripts/moveDmp.pl:139]: If the core is not in the preferred location, move it to $moveLocation

**Comment by llxia** [comment]
- Not only we need to test on mac, but we also need to test on wins, aix, xlinux and zos
- test a passed test
- test in both internal and Adopt Jenkins (selected platforms)

**Comment by LongyuZhang** [comment]
So far, following tests have passed: 
Archived core on Internal mac, xlinux, win, aix, zos are Grinder_TKG/378-381, 389
Archived core on Adopt xlinux, win, aix are https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/1226 - 1228
One passed test on Internal mac, xlinux, win, aix, zos are Grinder_TKG/406-409, 414
One passed test on Adopt mac, xlinux, win, aix are https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/1256 - 1259

The only problem remaining is core generate on Adopt Jenkins Mac machine met permission issue and cannot generate, posted in Issue https://github.com/AdoptOpenJDK/openjdk-tests/issues/1564 for more information.

**Comment by LongyuZhang** [comment]
@llxia This PR has been updated to remove $spec, and re-tested on the following platforms (Adopt Mac is still waiting for permission change), thanks.
Grinder_TKG:
	aix: 583   xlinux: 584    win:593   zos:600   mac:592
Adopt:
  aix: https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/1287/
  xlinux: https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/1288/
  win: https://ci.adoptopenjdk.net/view/work%20in%20progress/job/Grinder_Sandbox/1310
  mac: Still waiting for permission change in issue https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1282

**Pull request merged**